base/weakkeydict.jl

```diff
@@ -2,6 +2,12 @@
 
 # weak key dictionaries
 
+struct WeakKeyDictFinalizer{T}
```

Comment:
```suggestion
mutable struct WeakKeyDictFinalizer{T}
```

Reply:
why does this need to be mutable?

Reply:
the point of not declaring the field before was to ensure the compiler would box this object. making it mutable will do the same thing though